core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +70,45 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  @FunctionalInterface
+  private interface PropertyResolver<A, B, C, R> {
+    R apply(A argA, B argB, C argC);
+  }
+
+  private static <T> T resolveProperty(PropertyResolver<Map<String, String>, String, T, T> resolver,
+                                    Map<String, String> options, String optionKey,
+                                    Map<String, String> properties, String propertyKey, T defaultVal) {
+    return resolver.apply(options, optionKey, resolver.apply(properties, propertyKey, defaultVal));
+  }
+
+  private static <T> T resolveProperty(PropertyResolver<Map<String, String>, String, T, T> resolver,
+                                      Map<String, String> options, String optionKey,
+                                      Map<String, String> properties1, String propertyKey1,
+                                      Map<String, String> properties2, String propertyKey2, T defaultVal) {
```

Comment:
Looks like indentation is off here and the rest of the file is using double continuation indents.